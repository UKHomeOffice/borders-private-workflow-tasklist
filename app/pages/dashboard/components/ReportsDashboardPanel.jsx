import React from "react";
import {withRouter} from "react-router";
import AppConstants from "../../../common/AppConstants";

class ReportsDashboardPanel extends React.Component {


    reports(e) {
        e.preventDefault();
        this.props.history.replace({
            pathname: AppConstants.REPORTS_PATH,
            shiftPresent: this.props.hasActiveShift
        });
    }

    render() {
        return <li className="__card column-one-third" id="reportsPanel">
            <a href="#" onClick={this.reports.bind(this)} className="card__body">
                <span className="bold-large">reports</span>
            </a>
            <div className="card__footer">
                <span className="font-small">Operational reports</span>
            </div>
        </li>
    }
}

export default withRouter(ReportsDashboardPanel);