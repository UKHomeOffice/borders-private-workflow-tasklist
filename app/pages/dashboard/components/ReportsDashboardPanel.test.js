import React from 'react';
import { mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router-dom';
import { ReportsDashboardPanel } from './ReportsDashboardPanel';

describe('Reports Dashboard Panel', () => {
  const mockStore = configureStore();
  let store;
  const initialState = {
    'calendar-page': {},
  };

  beforeEach(() => {
    store = mockStore(initialState);
  });

  it('renders reports dashboard panel', async () => {
    const props = {};
    const wrapper = await mount(<ReportsDashboardPanel
      store={store}
      {...props}
    />);
    expect(wrapper).toMatchSnapshot();
  });
});
