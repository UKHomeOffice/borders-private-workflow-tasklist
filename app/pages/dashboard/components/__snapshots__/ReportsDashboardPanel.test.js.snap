// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reports Dashboard Panel renders reports dashboard panel 1`] = `
<ReportsDashboardPanel
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <li
    className="__card column-one-third"
    id="reportsPanel"
  >
    <a
      className="card__body"
      href="#"
      id="reportsPageLink"
      onClick={[Function]}
    >
      <span
        className="bold-large"
      >
        reports
      </span>
    </a>
    <div
      className="card__footer"
    >
      <span
        className="font-small"
      >
        Operational reports
      </span>
    </div>
  </li>
</ReportsDashboardPanel>
`;
